### PR TITLE
ADBDEV-2135 ADB 5.29.0 sync

### PR DESCRIPTION
**Pull request by deart2k**
## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Review by Stolb27 — APPROVED** [review]
tests will fail until we don't merge #231 and #254. I'm going to merge both with rebasing after this one.

@darthunix, @maksm90 please take a look at this one in the morning.

**Review by darthunix — APPROVED** [review]

**Pull request merged**
